### PR TITLE
Feature/abd timer persist proptypes

### LINKED ISSUE #104: Add proptypes to components that are using `toggleTimerView` method
Add typechecking for new method being passed down as props

### PR DESCRIPTION
**Pull request by Abdeboskey**
# Som Timer Pull Request Form

## Description
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

* This lil baby branch adds typechecking for the newly added `toggleTimerView` method that is passed as props to the children of `App`

* This branch also converts the `if/else` statement in the `toggleTimerView` method back to a ternary operator based on discussion and Som Timer company convention.

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other, Please explain:

* This PR resolves #104 

## Has This Been Tested?
- [ ] No
- [x] Yes
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration:

* The test suite was run to ensure that there would be no issues with CI due to these changes.

# Message for reviewer:

* You're doing a great job, and should feel real good about yourself.

# Checklist:
- [x] My code has no unused/commented out code
- [x] My code has no console logs
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/Components/PageNotFound/PageNotFound.js [hunk 1]
```diff
@@ -35,4 +36,8 @@ const PageNotFound = ({ toggleTimerView }) => {
   );
 }
 
-export default PageNotFound
\ No newline at end of file
+export default PageNotFound
+
+PageNotFound.propTypes = {
```
Comment (jkwest-93) [inline src/Components/PageNotFound/PageNotFound.js:41]: Thanks for adding this in!! We ❤️ type checking

**Review by jkwest-93 — APPROVED** [review]

File: src/Components/App/App.js [hunk 2]
```diff
@@ -16,11 +16,9 @@ function App() {
 
   const toggleTimerView = (hide) => {
     const homeContainer = document.getElementById('HomeContainer')
-    if (hide) {
-      homeContainer.classList.add('hidden')
-    } else {
-      homeContainer.classList.remove('hidden')
-    }
+    hide
```
Comment (rwilliams659) [inline src/Components/App/App.js:19]: Sorry I'm being soo picky again, but I think convention for ternaries based on previous threads I was reading about ternaries v if/else is that the "?" should live on the first line - better for dev empathy as it's easier to recognize you're reading a ternary as soon as you see that "?" (E.g. here is the ES Lint guideline: https://eslint.org/docs/rules/multiline-ternary)

File: src/Components/Settings/Settings.js [hunk 3]
```diff
@@ -87,15 +88,15 @@ const Settings = ({ toggleTimerView }) => {
       <div className={style.line}></div>
       <div className={style.intervalContainer}>
         <p className={style.intervalLabel}>{settings.moodRating ? 'Disable' : 'Enable'} mood ratings</p>
-        <label class={style.switch}>
+        <label className={style.switch}>
```
Comment (rwilliams659) [inline src/Components/Settings/Settings.js:91]: Good catch here!

File: src/Components/Stats/Stats.js [hunk 4]
```diff
@@ -30,4 +31,8 @@ const Stats = ({ toggleTimerView }) => {
   );
 }
 
-export default Stats
\ No newline at end of file
+export default Stats
+
+Stats.propTypes = {
```
Comment (rwilliams659) [inline src/Components/Stats/Stats.js:36]: Thanks for adding in all the propTypes!

**Review by rwilliams659 — APPROVED** [review]
Thanks for doing this :) Just left one little baby comment to address

File: src/Components/App/App.js (hunk 2 above)
Comment (Abdeboskey) [reply src/Components/App/App.js:19]: Aha,interesting to see it this way in the es-lint docs! I formatted this the other way based on the airBnB style guide that I referenced in our previous discussion and other examples I saw in my research. But ultimately I think that we should do what makes most sense to us, and if there's a strong preference plz let me know!

**Review by sciencefixion — APPROVED** [review]

**Pull request merged**